### ISSUE TITLE
SNP browser to replace Panoptes

### ISSUE THREAD

**Issue by EricRLucas** [malariagen/malariagen-data-python#563]
I tried to access the Panoptes web brower today, and found that it is being decommissioned. 

I gazed upon its ruin, and I wept. 

I gazed upon it some more, and I did wail, and beat my chest, and gnash my teeth. 

It would be good to have some way in malariagen_data to put in a bunch of snp position and get information on snp effect, allele frequencies, and a plot with the genomic context for the snps. Something along the lines of the colab below, but less crap. 

To the extent that there may be gems within with sewage, you are welcome to reuse that notebook. Also happy to help if needed. 

https://colab.research.google.com/drive/1e8AI2PjZgIWLJyciz4h6TbYF4B_PBe59?usp=drive_link

**Comment by leehart** [comment]
Ref: https://github.com/cggh/panoptes

**Comment by alimanfoo** [comment]
Thanks @EricRLucas, this is the best issue that's ever been raised on GitHub 😆

**Comment by sanjaynagi** [comment]
actually needed something like this the other day - would you be interested in adding it? @EricRLucas

**Comment by vigg-lstm** [comment]
@sanjaynagi  I'm happy to help, but certainly couldn't do it myself as I don't know enough about the workings of malariagen data. In the meantime, feel free to just make a copy of the colab I posted in the original issue. Should hopefully work out of the box

**Comment by EricRLucas** [comment]
Oops, looks like I was logged in with the vigg-lstm account when writing the above

**Comment by leehart** [comment]
@alimanfoo Maybe I should make a proof-of-concept PR based on @EricRLucas's Colab notebook, then we can all help to turn it into something beautiful, to all eyes. By the hammer of Hephaestus and the caduceus of Hermes...

**Comment by leehart** [comment]
A local prototype is alive:
<img width="583" alt="Screenshot 2024-10-01 at 21 53 15" src="https://github.com/user-attachments/assets/065ee858-f324-4618-8380-d8db06311a46">

**Comment by EricRLucas** [comment]
Very niice, thank you @leehart !  Is it possible for the function to also return the dataframe from the notebook? Or to have a separate function to do this (and perhaps even display it with the cells highlighted as in the notebook?)

**Comment by leehart** [comment]
> have a separate function to do this (and perhaps even display it with the cells highlighted as in the notebook?)

@EricRLucas Yep, that's already in the works. I'll show that tomorrow. I'll also try to get the code into a good enough state for a PR soon.

**Comment by leehart** [comment]
Plan to break into smaller issues.

**Comment by vigg-lstm** [comment]
Thanks Jon. Looks good to me, although I also think there is value in my original approach that looks at all the transcripts and flags the most "serious" effect (so maybe it's in an intron in the canonical transcript, but it's a non-synonymous mutation in one of the other transcripts, then it gets flagged as a non-synonymous mutations.

**Comment by mohamed-laarej** [comment]
Hi @jonbrenas ,

Thank you for the detailed breakdown and for tasking me with helping address some of these sub-issues. I'm very happy to contribute to improving the API.
I will take a deeper look into the issue and the proposed functions to ensure I understand everything thoroughly. Please let me know if there's anything specific you'd like me to focus on initially.

**Comment by jonbrenas** [comment]
> Thanks Jon. Looks good to me, although I also think there is value in my original approach that looks at all the transcripts and flags the most "serious" effect (so maybe it's in an intron in the canonical transcript, but it's a non-synonymous mutation in one of the other transcripts, then it gets flagged as a non-synonymous mutations.

I knew I was forgetting something. Thank you, I added it to the list.
